toolbox/CMakeLists.txt

```diff
@@ -202,57 +221,136 @@ configure_block(BLOCK_NAME "Get Limits"
 get_property(ALL_HEADERS GLOBAL PROPERTY WBT_HEADERS)
 get_property(ALL_SOURCES GLOBAL PROPERTY WBT_SOURCES)
 
-add_definitions(-DMATLAB_MEX_FILE)
-include_directories(include)
-include_directories(include/base)
+add_library(WBToolboxLibrary SHARED
+    ${ALL_HEADERS} ${ALL_SOURCES}
+)
+set_target_properties(WBToolboxLibrary PROPERTIES
+    OUTPUT_NAME "WBToolbox"
+    # VERSION ${PROJECT_VERSION}
+    PUBLIC_HEADER "${ALL_HEADERS}"
+)
+
+# Import math symbols from standard cmath
+target_compile_definitions(WBToolboxLibrary PUBLIC "-D_USE_MATH_DEFINES")
```

Comment:
Why `PUBLIC` ? I far as I can see, `M_PI` is used only in `.cpp` files. I would turn it `PRIVATE` and only add it only when compiling on `MSVC` ?

Reply:
Good catch!